integration/kubernetes/helm-chart/alluxio/templates/master/statefulset.yaml

```diff
@@ -80,160 +74,135 @@ spec:
       {{- end}}
       containers:
         - name: alluxio-master
-          image: {{ $root.Values.image }}:{{ $root.Values.imageTag }}
-          imagePullPolicy: {{ $root.Values.imagePullPolicy }}
+          image: {{ .Values.image }}:{{ .Values.imageTag }}
+          imagePullPolicy: {{ .Values.imagePullPolicy }}
           securityContext:
-            runAsUser: {{ $root.Values.user }}
-            runAsGroup: {{ $root.Values.group }}
-          {{- if $root.Values.master.resources  }}
-          resources:
-            {{- if $root.Values.master.resources.limits }}
-            limits:
-              cpu: {{ $root.Values.master.resources.limits.cpu }}
-              memory: {{ $root.Values.master.resources.limits.memory }}
-            {{- end }}
-            {{- if $root.Values.master.resources.requests }}
-            requests:
-              cpu: {{ $root.Values.master.resources.requests.cpu }}
-              memory: {{ $root.Values.master.resources.requests.memory }}
-            {{- end }}
+            runAsUser: {{ .Values.user }}
+            runAsGroup: {{ .Values.group }}
+          {{- if .Values.master.resources  }}
+{{ include "alluxio.master.resources" . | indent 10 }}
           {{- end }}
           command: ["/entrypoint.sh"]
-          {{- if $root.Values.master.args }}
+          {{- if .Values.master.args }}
           args:
-{{ toYaml $root.Values.master.args | trim | indent 12 }}
+{{ toYaml .Values.master.args | trim | indent 12 }}
           {{- end }}
           {{- if $isHaEmbedded }}
           env:
-          - name: ALLUXIO_MASTER_JAVA_OPTS
-            value: {{ $masterJavaOpts | quote }}
+          - name: ALLUXIO_MASTER_HOSTNAME
+              valueFrom:
+                fieldRef:
+                  fieldPath: metadata.name
           - name: ALLUXIO_USER_JAVA_OPTS
-            value: {{ $masterJavaOpts | quote }}
+            value: " -Dalluxio.master.hostname=${ALLUXIO_MASTER_HOSTNAME} "
```

Comment:
i'm confused about this, although it isn't related to this PR. 

for the embedded journal configuration the client does not need `alluxio.master.hostname` but instead `alluxio.master.embedded.journal.addresses` should be set. move `ALLUXIO_USER_JAVA_OPTS ` to the configMap as well?

Reply:
For `ALLUXIO_USER_JAVA_OPTS`, it used to only have `alluxio.master.hostname` from the beginning. In my initial helm change I parameterized into using parameter `$masterJavaOpts`, which only had `-Dalluxio.master.hostname=alluxio-master-{index}`. Now that we don't need index anymore I changed it back to the hostname, in this PR.

Back to your question, `alluxio.master.embedded.journal.addresses` is always in ALLUXIO_JAVA_OPTS, while `alluxio.master.hostname` is only in ALLUXIO_MASTER_JAVA_OPTS and ALLUXIO_JOB_MASTER_JAVA_OPTS. So I think the client will have the correct properties?

However I agree `ALLUXIO_USER_JAVA_OPTS` should move to the configMap. Will do it in another commit.